Lib/selectors.py

```diff
@@ -252,15 +252,14 @@ def unregister(self, fileobj):
         return key
 
     def modify(self, fileobj, events, data=None):
-        # TODO: Subclasses can probably optimize this even further.
         try:
             key = self._fd_to_key[self._fileobj_lookup(fileobj)]
         except KeyError:
             raise KeyError("{!r} is not registered".format(fileobj)) from None
         if events != key.events:
             self.unregister(fileobj)
             key = self.register(fileobj, events, data)
-        elif data != key.data:
+        elif data is not key.data:
```

Comment:
I disagree with this change. I've found nothing in the docs constraining the `__eq__` sematics of data. In the common cases, where data is either None or unchanged, `__eq__` typically still takes a shortcut when comparing the object to itself. So the story about comparing two 1Gb strings seems unlikely -- either it's the same string, in which case it takes a shortcut, or it's two different strings, in which case the comparison bails at the first difference. In case the data is e.g. a list, and the new data is a different list containing the same values, this "optimization" would actually be a slow-down. In general it smells superstitious. I think the code should show the intended semantics.